compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4618,7 +4618,8 @@ private STNode createMissingSimpleObjectField(STNode metadata, List<STNode> qual
     }
 
     private STNode modifyNodeWithInvalidTokenList(List<STNode> qualifiers, STNode node) {
-        for (STNode qualifier : qualifiers) {
+        for (int i = qualifiers.size() - 1; i >= 0; i--) {
+            STNode qualifier = qualifiers.get(i);
```

Comment:
R we having the same issue with `reportInvalidQualifierList` as well?

Reply:
Inside `reportInvalidQualifierList()` we call `addInvalidNodeToNextToken()` for each token, which keeps the tokens in a stack. Since stack is FILO order we get the right order there.

Reply:
Cool